druntime/test/shared/src/lib.d

```diff
@@ -1,5 +1,11 @@
 module lib;
 
+// use _assert instead of assert to also fail in release builds
+void _assert(bool cond)
+{
+    if (!cond) assert(false);
+}
+
```

Comment:
I'm wondering whether we shouldn't just remove the `-release` in https://github.com/dlang/dmd/blob/3173112dd3907d02ed9e74ba9e9f6047c1725d5a/druntime/test/common.mak#L36

For LDC, I've similarly added `-check=assert=on` in https://github.com/ldc-developers/ldc/blob/1d4c2cfc5e17e6cfcf7cd154d5e66394162fa981/runtime/druntime/test/profile/Makefile#L26.